server/icon.go

```diff
@@ -0,0 +1,235 @@
+// Copyright (c) 2025-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"image"
+	_ "image/png"
+	"io"
+	"net/http"
+	"strings"
+
+	"github.com/gorilla/mux"
+	"github.com/mattermost/mattermost-plugin-msteams-devsecops/assets"
+	"github.com/mattermost/mattermost-plugin-msteams-devsecops/server/store/pluginstore"
+	"github.com/mattermost/mattermost/server/public/model"
+	"github.com/sirupsen/logrus"
+)
+
+type IconType string
+
+const (
+	IconTypeColor   IconType = "color"
+	IconTypeOutline IconType = "outline"
+)
+
+// IsValid checks if the IconType is valid
+func (it IconType) IsValid() bool {
+	return it == IconTypeColor || it == IconTypeOutline
+}
+
+// uploadIcon handles file uploads for custom icons
+func (a *API) uploadIcon(w http.ResponseWriter, r *http.Request) {
```

Comment:
I guess since this is an admin endpoint we don't need to check the file type using the file header nor use a limted reader. cc: @enzowritescode